tex2wiki/src/tex2wiki.py

```diff
@@ -0,0 +1,413 @@
+    Special case of generate_html, where the tag is "math".
+    """
+    if options is None:
+        options = {}
+
+    option_text = [key + "=\"" + value + "\"" for key, value in options.iteritems()]
+    result = "<math" + " " * (option_text != []) + ", ".join(option_text) + ">"
+
+    if spacing:
+        result += "{\\displaystyle\n" + text + "\n}</math>\n"
+    else:
+        result += "{\\displaystyle " + text + "}</math>\n"
+
+    return result
+
+
+def generate_link(left, right=""):
+    """Generates a link thingie."""
+
+    if right == "":
+        return "[[" + left + "|" + left + "]]"
+
+    return "[[" + left + "|" + right + "]]"
+
+
+def multi_split(s, seps):
+    """Copy pasted from internet!"""
+    res = [s]
+    for sep in seps:
+        s, res = res, []
+        for seq in s:
+            res += seq.split(sep)
+    return res
+
+
+def convert_dollar_signs(string):
+    count = 0
+    result = ""
+    for i, ch in enumerate(string):
+        if ch == "$" and count % 2 == 0:
+            result += "<math>{\\displaystyle "
+            count += 1
+        elif ch == "$":
+            result += "}</math>"
+            count += 1
+        else:
+            result += ch
+
+    return result
+
+
+def format_formula(formula):
+    if formula[0] == ":":
+        return formula[1:].zfill(2)
+
+    formula = multi_split(formula.split("Formula:", 1)[1], [".", ":"])
+
+    for j in [-1, -2, -3]:
+        formula[j] = formula[j].zfill(2)
+
+    if len(formula) == 3:
+        formula = formula[0] + "." + formula[1] + ":" + formula[2]
+    else:
+        formula = ":".join(formula[:-3]) + ":" + formula[-3] + "." + formula[-2] + ":" + formula[-1]
+
+    return formula
+
+
+def format_metadata(string):
+    if string == "":
+        return ""
+
+    # modified convert_dollar_sign algorithm
+    dollar_count = 0
+    result = ""
+    for i, ch in enumerate(string):
+        if ch == "$":
+            if dollar_count % 2 == 1:
+                result += "}</math>"
+            elif dollar_count > 1:
+                result = result[:-1] + "<br />\n<math>{\\displaystyle "
+            else:
+                result += "<math>{\\displaystyle "
+            dollar_count += 1
+        else:
+            result += ch
+
+    return result.strip()
+
+
+def extract_data(data):
+    result = list()
+    for section_data in data:
+        equations = section_data[1][:-1]
+        for i, equation in enumerate(equations):
+            # formula stuff
+            try:
+                formula = format_formula(get_data_str(equation, latex="\\formula"))
+            except IndexError:  # there is no formula.
+                break
+
+            equation = equation.split("\n")[1:]
+
+            # get metadata
+            percent_list = list()
+            raw_metadata = ""
+
+            j = 0
+            while j < len(equation):
+                if "%" in equation[j]:
+                    raw_metadata += equation.pop(j)[1:].strip().strip("\n") + "\n"
```

Comment:
`str.strip()` strips _whitespace_, not spaces. In other words, the `.strip("\n")` is unnecessary because `.strip()` already did that.